File: seastarvideo/src/main/java/com/freewind/seastarvideo/api/Callback.java
package com.freewind.seastarvideo.api;

import com.freewind.seastarvideo.bean.BaseBean;

public abstract class Callback<T extends BaseBean> implements ApiImpl<T> {
    @Override
    public void onStart() {

    }

    @Override
    public void onCancel() {

    }

    @Override
    public void onSuccess(T data) {

    }

    @Override
    public void onFailed(int code, String msg) {

    }
}

File: seastarvideo/src/main/java/com/freewind/seastarvideo/http/Callback.kt
package com.freewind.seastarvideo.http

import com.freewind.seastarvideo.http.bean.BaseBean

abstract class Callback<T : BaseBean> : ApiImpl<T> {
    override fun onStart() {

    }

    override fun onCancel() {

    }

    override fun onSuccess(data: T) {

    }

    override fun onFailed(code: Int, msg: String?) {

    }
}
